src/thor/astar.cc

```diff
@@ -291,7 +291,7 @@ AStarPathAlgorithm::GetBestPath(valhalla::Location& origin,
     // invalid label indicates there are no edges that can be expanded.
     uint32_t predindex = adjacencylist_->pop();
     if (predindex == kInvalidLabel) {
-      LOG_ERROR("Route failed after iterations = " + std::to_string(edgelabels_.size()));
+      LOG_ERROR("Route failed after iterations = " + std::to_string(edgelabels_.size()) + " because of invalid edge-index");
```

Comment:
Invalid edge index is not really relevant to the error message. The comment above says exactly what it means. Basically invalid edge index just means its the adjacency list is empty, ie the search is exhausted and there arent any more edges to look at. I would favor a message more along those lines.